test/inject.test.js

```diff
@@ -409,3 +409,18 @@ test('should throw error if callback specified and if ready errors', t => {
     t.strictEqual(err, error)
   })
 })
+
+test('should support builder-style injection with non-ready app', async (t) => {
+  t.plan(3)
+  const fastify = Fastify()
+  const payload = { hello: 'world' }
+
+  fastify.get('/', (req, reply) => {
+    reply.send(payload)
+  })
+
+  const res = await fastify.inject().get('/').end()
+  t.deepEqual(payload, JSON.parse(res.payload))
+  t.strictEqual(res.statusCode, 200)
+  t.strictEqual(res.headers['content-length'], '17')
+})
```

Comment:
Tests are failing on Node 6 because there is no async-await there.

Reply:
Good to know, thanks!